### PR TITLE
add max ttl option when mount backend

### PR DESCRIPTION
**Pull request by danielzhanghl**
this code is to add  max ttl option when mount a backend

### CODE REVIEW

**Comment by danielzhanghl** [comment]
i am newer to add code to github, and in this request always get continuous-integration build failure as below,  did I miss anything?
thanks,

-----------
The command "ls -la build" exited with 0.
cache.2
store build cache
0.00s
1.65schanges detected, packing new archive
uploading archive
after_success.1
0.00s$ export PR=https://api.github.com/repos/$TRAVIS_REPO_SLUG/pulls/$TRAVIS_PULL_REQUEST
after_success.2
0.23s$ export BRANCH=$(if [ "$TRAVIS_PULL_REQUEST" == "false" ]; then echo $TRAVIS_BRANCH; else echo `curl -s $PR | jq -r .head.ref`; fi)
$ if [[ "$TRAVIS_PULL_REQUEST" != "false" ]]; then make -j docker DOCKER_USER=$DOCKER_USER DOCKER_PASS=$DOCKER_PASS TAG=pr-$TRAVIS_PULL_REQUEST COMMIT=$TRAVIS_COMMIT; fi
=> build and push Docker image ...
Password: 
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
The build has been terminated

**Comment by jippi** [comment]
I think this PR isn't necessary, you can do this already 

```hcl
environment "production" {
  mount "pki" {
    type = "pki"

    config "" {
      max-lease-ttl  = "87600h"
    }
  }
}
```

**Comment by danielzhanghl** [comment]
I configured as this way, but get below error, 
seem that need a name, and try to write "config" section to /pki/config/ always...


environment "production" {

  # the name/path the mount will be mounted at,
  # must be unique for the environment
  mount "pki" {

    # the mount backend type to use (see Vault docs)
    type = "pki"

    # mount configuration, see Vault docs for details
    config "urls" {
        issuing_certificates = "https://172.17.0.2:8800/v1/pki/ca"
        crl_distribution_points = "https://172.17.0.2:8800/v1/pki/crl"
    }

    config "" {
        max-lease-ttl  = "87600h"
    }

  }
}



FATA[0000] Error making API request.

URL: PUT https://172.17.0.2:8800/v1/pki/config
Code: 404. Errors:

* 1 error occurred:

* unsupported path

**Comment by jippi** [comment]
Interesting, can't find any config at https://www.vaultproject.io/api/secret/pki/index.html for `max-lease-ttl` though :o

**Comment by danielzhanghl** [comment]
you could use mount-tune to tune it, or set that when mount it, actually, that is general option to any backend.

File: config/parse_vault_mount.go [hunk 1]
```diff
@@ -16,7 +15,7 @@ func (c *Config) processVaultMounts(list *ast.ObjectList, environment *Environme
 	for _, mountAST := range list.Items {
 		x := mountAST.Val.(*ast.ObjectType).List
 
-		valid := []string{"config", "role", "type", "path"}
+		valid := []string{"config", "role", "type", "path", "maxleasettl"}
```
Comment (jippi) [inline config/parse_vault_mount.go]: can we call the hcl key for `max_lease_ttl`? :)

Comment (jippi) [inline config/parse_vault_mount.go]: oh, and add `default_lease_ttl` and `force_no_cache` that should cover all 3 options `MountConfigInput` covers - if you don't got time, i can do that

Comment (danielzhanghl) [inline config/parse_vault_mount.go]: ok, please do that, and it will be much better to update the example to include these  as well.
thanks!

**Comment by jippi** [comment]
Closes in favour of #17

**Pull request closed without merging**